unit_tests/src/test_rc_switch_user.cpp

```diff
@@ -282,9 +305,122 @@ TEST_CASE("rc_switch_user cannot be invoked by non-admins")
+
+    // Become the test user and instruct the server to update the proxy user as well.
+    auto* conn_ptr = static_cast<RcComm*>(conn);
+    SwitchUserInput input{};
+    std::strncpy(input.username, alice.name.c_str(), sizeof(SwitchUserInput::username));
+    std::strncpy(input.zone, alice.zone.c_str(), sizeof(SwitchUserInput::zone));
+    input.update_proxy_user = 1;
+    REQUIRE(rc_switch_user(conn_ptr, &input) == 0);
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    CHECK(conn_ptr->clientUser.userName == alice.name);
+    CHECK(conn_ptr->clientUser.rodsZone == alice.zone);
+    // NOLINTEND(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+
+    // Show that because SwitchUserInput::update_proxy_user is 1, iRODS no longer allows
+    // the rc_switch_user to be invoked.
+    std::strncpy(input.username, env.rodsUserName, sizeof(SwitchUserInput::username));
+    std::strncpy(input.zone, env.rodsZone, sizeof(SwitchUserInput::zone));
+    REQUIRE(rc_switch_user(conn_ptr, &input) == SYS_PROXYUSER_NO_PRIV);
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    CHECK(conn_ptr->clientUser.userName == alice.name);
+    CHECK(conn_ptr->clientUser.rodsZone == alice.zone);
+    // NOLINTEND(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+
+    // Become the administrator so that the test can clean up properly.
+    // This can only be achieved by reconnecting to the server due to the test user not
+    // being an administrator.
+    conn.disconnect();
+    conn.connect();
+}
+
+TEST_CASE("rc_switch_user can be called multiple times when update_proxy_user is 0")
+{
+    //
+    // IMPORTANT: This test requires access to a rodsadmin user!
+    //
+
+    load_client_api_plugins();
+
+    rodsEnv env;
+    _getRodsEnv(env);
+
+    // This connection is used for creating the test user and cleaning up.
+    irods::experimental::client_connection conn;
+
+    // Create two test rodsusers.
+    const adm::user alice{"test_user_alice", env.rodsZone};
+    REQUIRE_NOTHROW(adm::client::add_user(conn, alice));
+    REQUIRE_NOTHROW(adm::client::modify_user(conn, alice, adm::user_password_property{"rods"}));
+
+    irods::at_scope_exit remove_test_user_alice{[&conn, &alice] { REQUIRE_NOTHROW(adm::client::remove_user(conn, alice)); }};
+
+    const adm::user bob{"test_user_bob", env.rodsZone};
+    REQUIRE_NOTHROW(adm::client::add_user(conn, bob));
+    REQUIRE_NOTHROW(adm::client::modify_user(conn, bob, adm::user_password_property{"rods"}));
+
+    irods::at_scope_exit remove_test_user_bob{[&conn, &bob] { REQUIRE_NOTHROW(adm::client::remove_user(conn, bob)); }};
+
+    //
+    // Call rc_switch_user multiple times and toggle between two rodsusers.
+    // It is important that the users being switched to are rodsusers because doing this
+    // with rodsadmins may bypass important permission checks.
```

Comment:
consider groupadmins too - even if just in the comment... (as i'm sure they're the same as rodsusers for the purposes of this testing)

Reply:
Will do. I can add a check for groupadmins too.

Reply:
done

Reply:
Made one of the users a groupadmin.

Now the test toggles between a rodsuser and groupadmin.